engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/consumerrecord/ConsumerRecordToJsonFormatter.scala

```diff
@@ -0,0 +1,88 @@
+package pl.touk.nussknacker.engine.kafka.consumerrecord
+
+import java.nio.charset.{Charset, StandardCharsets}
+
+import com.github.ghik.silencer.silent
+import io.circe.Decoder.Result
+import io.circe.{Decoder, Encoder, HCursor, Json}
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.common.record.TimestampType
+import pl.touk.nussknacker.engine.api.CirceUtil
+import pl.touk.nussknacker.engine.api.test.{TestDataSplit, TestParsingUtils}
+import pl.touk.nussknacker.engine.kafka.{ConsumerRecordUtils, RecordFormatter}
+import pl.touk.nussknacker.engine.util.json.BestEffortJsonEncoder
+
+import scala.annotation.nowarn
+
+@silent("deprecated")
+@nowarn("cat=deprecation")
+class ConsumerRecordToJsonFormatter extends RecordFormatter {
+
+  private val cs: Charset = StandardCharsets.UTF_8
+
+  // TODO: add better key-value encoding and decoding
+  implicit val consumerRecordEncoder: Encoder[ConsumerRecord[Array[Byte], Array[Byte]]] = {
+    val encode: Encoder[Any] = BestEffortJsonEncoder(failOnUnkown = false).circeEncoder
+    new Encoder[ConsumerRecord[Array[Byte], Array[Byte]]] {
+      override def apply(a: ConsumerRecord[Array[Byte], Array[Byte]]): Json = Json.obj(
```

Comment:
I aggree that we should just encode/decode only fields that are used in `InputMeta`.  If it can be handled by json semiauto devriation, it would be perfect. Be aware @Grzesiek, that this jsons sometimes will be hand-written by users - if they need to prepare own tests inputs.

Reply:
Fixed (and introduced case class `SerializableConsumerRecord`)